### PR TITLE
Redis: added basic ability to replace cache adapter

### PR DESCRIPTION
**Pull request by bryanlatten**
- Note: unit tests are memcache/memcached-version sensitive, which is infuriating.

### CODE REVIEW

File: src/Adapters/RedisAdapter.php [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    return $this->_connection->close();
+
+  } // _close
+
+
+  /**
+   * Ensures legacy memcache clients using 0 for "permanent" become some arbitrarily large value, like a month
+   *
+   * @param int
```
Comment (TJP85) [inline src/Adapters/RedisAdapter.php]: `@param int $ttl`

File: tests/unit/Adapters/RedisAdapterTest.php [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+   */
+  public function passthru( $method, $translated, $args, $with, $expected ) {
+
+    $redis = $this->createMock( $this->_cache );
+
+    if ( $with === null ) {
+
+      $redis->expects( $this->once() )
+        ->method( $translated )
+        ->will( $this->returnValue( $expected ) );
```
Comment (TJP85) [inline tests/unit/Adapters/RedisAdapterTest.php]: `willReturn`

File: tests/unit/Adapters/RedisAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    return [
+        [ 'get',         'get',         [ $key ],            $key,  $value ],
+        [ 'increment',   'incrBy',      [ $key, $value, 1 ], null, true ],
+        [ 'decrement',   'decrBy',      [ $key, $value, 1 ], null, true ],
+        [ 'delete',      'delete',      [ $key ],            $key,  true ],
+        [ 'deleteMulti', 'delete',      [ $keys ],           $keys, true ],
+        [ 'flush',       'flushDb',     [],                  null, true ],
+        [ 'close',       'close',       [],                  null, true ],
+        // [ 'getStats',    'info',        [],                      null, $keys ],
+        // [ 'getAllKeys',  'scan',        [],                      null, $keys ],
```
Comment (TJP85) [inline tests/unit/Adapters/RedisAdapterTest.php]: need this?

**Comment by TJP85** [comment]
👍

**Comment by mikeklein13** [comment]
👍

**Pull request merged**